cmd/aks-periscope/aks-periscope.go

```diff
@@ -42,6 +43,21 @@ func main() {
 	systemPerfCollector := collector.NewSystemPerfCollector(exporter)
 	collectors = append(collectors, systemPerfCollector)
 
+	// Checks flags and adds specified collector if matched
+	flags := strings.Fields(os.Getenv("FLAGS_LIST"))
+	isOsmEnabled := false
```

Comment:
💡 🐛 This will introduce a bug in your implementation logic here: What I read from below code is `OSM` and `SMI` are mutually exclusive flags? (because if `isOsmEnabled` is **true**, the subsequent flag for `SMI` will not reach because `&& !isOsmEnabled`

Further: 

* `isOsmEnabled := false` and then used for `if` conditions (I think these are the only places where this car is used)

* What is the use case of list getting passed with `OSM and SMI` together? In this case your `isOsmEnabled` is set to `true` already?

Reply:
@shalier

Reply:
>What is the use case of list getting passed with OSM and SMI together? In this case your isOsmEnabled is set to true already?

If the user accidentally sets both `OSM` and `SMI` flag, then we wanted to avoid appending the SMI collector twice to `collectors`. To prevent this I had the `isOsmEnabled` boolean, but I see there's an issue where it wouldn't catch double-appending if the user enables the flags in a different order `SMI OSM` 

If running the SMI collector twice isn't a concern we can remove it or I can work on it some more to prevent the SMI collector from running twice, lmk what you think @Tatsinnit?

Reply:
(some context to how these two flags interact) OSM applies SMI policies, so when the `OSM` collector is in use, we also want information about SMI policies being applied. However, SMI policies can be applied with other service meshes as well so the `SMI` flag ensures that users who want to collect just SMI information are able to do that regardless of which service mesh solution they are using.
> * What is the use case of list getting passed with OSM and SMI together?

Ideally this should never happen but could happen by mistake, hence the logic Shalier explained above

Reply:
@Tatsinnit Good point, we'll implement this way in the SMI pr